integration-tests/common/src/test/kotlin/me/tatarka/inject/test/InjectObjectTest.kt

```diff
@@ -0,0 +1,41 @@
+package me.tatarka.inject.test
+
+import me.tatarka.inject.annotations.Component
+import me.tatarka.inject.annotations.Inject
+import me.tatarka.inject.test.different.DifferentPackageFoo
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+
+@Inject object MyInjectObject
+
+@Inject class DependsOnMyInjectObject(val myInjectObject: MyInjectObject)
+
+@Component interface MyComponent {
+    val injectObject: MyInjectObject
+    val differentPackageObject: DifferentPackageFoo.MyObject
+    fun injectObject(): MyInjectObject
+    fun dependsOnMyInjectObject(): DependsOnMyInjectObject
+}
+
+class InjectObjectTest {
+    @Test
+    fun inject_annotated_object_can_be_provided_in_component_function() {
+        assertThat(InjectMyComponent().injectObject(), equalTo(MyInjectObject))
```

Comment:
Please use assertk assertions to match the rest of the tests, ex: `assertThat(InjectMyComponent().injectObject()).isEqualTo(MyInjectObject)`